### DISCUSSION TITLE
Arbitrary Values for width

### DISCUSSION THREAD

**Discussion by sanket-16** [tailwindlabs/tailwindcss#9408] [Help] [answered]
I'm making a progress bar for which I need the bar inside to be foo % of the parent div. For implementing this I used the method according to tailwind Docs i.e. `w-[78%]`. But for some reason the `w-[78%]` doesn't work , but the one's provided for them `w-5/6` which is 83% works. Also rems ,pixels don't seem to work either.

Adding style and then using {width: progress} works perfectly fine and smilarly I tried assigning value like `h-[80vh]` to a element which worked perfectly fine. 

Any help is appreciated..

JIT mode is on and tailwind version is 3.1

**Comment by wongjn** [comment, marked as answer]
The classes need to exist in your source files as full, uninterrupted strings at build time first, [as per the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as *complete unbroken strings* in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:

I like to think about it conceptually like Tailwind looks at your files as though they were `.txt` files. It sounds like you might be constructing your width class dynamically, like `w-[${progress}]` or something similar. Tailwind does not know about how your code runs (again, it reads your files like they are `.txt` files) and does not know that `w-[${percentage}]` could become `w-[78%]`. What you could do alternatively is – as you say – use `width` as an inline style.

**Reply by sanket-16** [reply]
Thank you!
